show_client/system_health_cli.go

```diff
@@ -0,0 +1,224 @@
+			OperStatus: operStatus,
+		}
+
+		// Find state, time, and reason fields for this state type
+		for key, value := range stateInfo {
+			if strValue, ok := value.(string); ok {
+				if strings.Contains(key, stateType) {
+					if strings.HasSuffix(key, "_state") {
+						row.StateDetail = key
+						row.StateValue = strValue
+						if strings.ToLower(strValue) == "up" {
+							row.Reason = ""
+						}
+					} else if strings.HasSuffix(key, "_time") {
+						row.Time = strValue
+					} else if strings.HasSuffix(key, "_reason") {
+						if strings.ToLower(row.StateValue) != "up" {
+							row.Reason = strValue
+						}
+					}
+				}
+			}
+		}
+
+		// Only add row if we have state information
+		if row.StateDetail != "" {
+			rows = append(rows, row)
+		}
+	}
+
+	return rows
+}
+
+func getSystemHealthDpu(options sdc.OptionMap) ([]byte, error) {
+	log.V(2).Infof("getSystemHealthDpu: called with options: %v", getOptionsKeys(options))
+
+	// Get module name from options
+	moduleName := "all" // default to all modules
+	if moduleStr, ok := options["dpu"].String(); ok {
+		moduleName = moduleStr
+		log.V(2).Infof("getSystemHealthDpu: filtering for module: %s", moduleName)
+	}
+
+	// Get data for specified module(s) using direct chassis database connection
+	log.V(2).Infof("getSystemHealthDpu: getting data for module: %s", moduleName)
+	stateData, err := getChassisDataDirect(moduleName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Process the data
+	var allRows []DpuStateRow
+	for moduleKey, stateInfo := range stateData {
+		// Extract module name from key (format: "DPU_STATE|DPU0")
+		keyParts := strings.Split(moduleKey, "|")
+		if len(keyParts) != 2 {
+			log.V(2).Infof("getSystemHealthDpu: skipping invalid key: %s", moduleKey)
+			continue
+		}
+		moduleName := keyParts[1]
+
+		stateInfoMap, ok := stateInfo.(map[string]interface{})
+		if !ok {
+			log.V(2).Infof("getSystemHealthDpu: skipping invalid state info for module: %s", moduleName)
+			continue
+		}
+
+		// Create rows for this module
+		rows := createDpuStateRowsFromData(moduleName, stateInfoMap)
+		allRows = append(allRows, rows...)
+	}
+
+	// Sort results by DPU name for consistent ordering
+	sort.Slice(allRows, func(i, j int) bool {
+		return allRows[i].Name < allRows[j].Name
+	})
+
+	// If no rows found, return not found error
+	if len(allRows) == 0 {
+		log.V(2).Infof("getSystemHealthDpu: module %s not found", moduleName)
+		return nil, status.Errorf(codes.NotFound, "module %s not found", moduleName)
+	}
+
+	// Convert to JSON
+	jsonData, err := json.Marshal(allRows)
+	if err != nil {
+		log.V(2).Infof("getSystemHealthDpu: error marshaling result: %v", err)
+		return nil, fmt.Errorf("failed to marshal result: %w", err)
+	}
+
+	log.V(2).Infof("getSystemHealthDpu: returning: %s", string(jsonData))
+	return jsonData, nil
+}
+
+// Get data directly from chassis database using SonicV2Connector
+func getChassisDataDirect(moduleName string) (map[string]interface{}, error) {
```

Comment:
can you explain why we need to use this instead of existing helper functions or even existing redis client? why do we have to reimplement db fetching logic?

Reply:
Will add in next pr.